sqlmesh/core/context_diff.py

```diff
@@ -202,77 +214,81 @@ def promotable_models(self) -> t.Set[str]:
         return current.fingerprint.data_hash != previous.fingerprint.data_hash
 
-    def indirectly_modified(self, model_name: str) -> bool:
-        """Returns whether or not a model was indirectly modified in this context.
+    def indirectly_modified(self, name: str) -> bool:
+        """Returns whether or not a node was indirectly modified in this context.
 
         Args:
-            model_name: The model name to check.
+            name: The node name to check.
 
         Returns:
-            Whether or not the model was indirectly modified.
+            Whether or not the node was indirectly modified.
         """
 
-        if model_name not in self.modified_snapshots:
+        if name not in self.modified_snapshots:
             return False
 
-        current, previous = self.modified_snapshots[model_name]
+        current, previous = self.modified_snapshots[name]
         return (
             current.fingerprint.data_hash == previous.fingerprint.data_hash
             and current.fingerprint.parent_data_hash != previous.fingerprint.parent_data_hash
         )
 
-    def metadata_updated(self, model_name: str) -> bool:
-        """Returns whether or not the given model's metadata has been updated.
+    def metadata_updated(self, name: str) -> bool:
+        """Returns whether or not the given node's metadata has been updated.
 
         Args:
-            model_name: The model name to check.
+            name: The node name to check.
 
         Returns:
-            Whether or not the model's metadata has been updated.
+            Whether or not the node's metadata has been updated.
         """
 
-        if model_name not in self.modified_snapshots:
+        if name not in self.modified_snapshots:
             return False
 
-        current, previous = self.modified_snapshots[model_name]
+        current, previous = self.modified_snapshots[name]
         return current.fingerprint.metadata_hash != previous.fingerprint.metadata_hash
 
-    def text_diff(self, model: str) -> str:
-        """Finds the difference of a model between the current and remote environment.
+    def text_diff(self, node: str) -> str:
+        """Finds the difference of a node between the current and remote environment.
 
         Args:
-            model: The model name.
+            node: The node name.
 
         Returns:
-            A unified text diff of the model.
+            A unified text diff of the node.
         """
-        if model not in self.snapshots:
-            raise SQLMeshError(f"`{model}` does not exist.")
-        if model not in self.modified_snapshots:
+        if node not in self.snapshots:
+            raise SQLMeshError(f"`{node}` does not exist.")
+        if node not in self.modified_snapshots:
             return ""
 
-        new, old = self.modified_snapshots[model]
-        return old.model.text_diff(new.model)
+        new, old = self.modified_snapshots[node]
+        if new.is_model:
```

Comment:
Shouldn't `text_diff` be a behavior of a snapshot or a node to avoid this branching?